### DISCUSSION TITLE
TSwap: make exits with error

### DISCUSSION THREAD

**Discussion by Himchenhim** [Cyfrin/security-and-auditing-full-course-s23#109] [Q&A]
After cloning "5-t-swap-audit" repository, I am trying to run "make" but it exits with the error:

❯ make
rm -rf .gitmodules && rm -rf .git/modules/* && rm -rf lib && touch .gitmodules && git add . && git commit -m "modules"
[main 674c4b6] modules
 3 files changed, 8 deletions(-)
 delete mode 160000 lib/forge-std
 delete mode 160000 lib/openzeppelin-contracts
forge install foundry-rs/forge-std --no-commit && forge install openzeppelin/openzeppelin-contracts --no-commit
Installing forge-std in /home/ziion/reports/tswap/5-t-swap-audit/lib/forge-std (url: Some("https://github.com/foundry-rs/forge-std"), tag: None)
Cloning into '/home/ziion/reports/tswap/5-t-swap-audit/lib/forge-std'...
remote: Enumerating objects: 2135, done.
remote: Counting objects: 100% (2131/2131), done.
remote: Compressing objects: 100% (699/699), done.
remote: Total 2135 (delta 1416), reused 2048 (delta 1365), pack-reused 4
Receiving objects: 100% (2135/2135), 591.36 KiB | 198.00 KiB/s, done.
Resolving deltas: 100% (1416/1416), done.
Submodule 'lib/ds-test' (https://github.com/dapphub/ds-test) registered for path 'lib/forge-std/lib/ds-test'
Cloning into '/home/ziion/reports/tswap/5-t-swap-audit/lib/forge-std/lib/ds-test'...
remote: Enumerating objects: 313, done.        
remote: Counting objects: 100% (171/171), done.        
remote: Compressing objects: 100% (87/87), done.        
remote: Total 313 (delta 90), reused 135 (delta 75), pack-reused 142        
Receiving objects: 100% (313/313), 71.98 KiB | 35.00 KiB/s, done.
Resolving deltas: 100% (129/129), done.
    Installed forge-std v1.7.6
Installing openzeppelin-contracts in /home/ziion/reports/tswap/5-t-swap-audit/lib/openzeppelin-contracts (url: Some("https://github.com/openzeppelin/openzeppelin-contracts"), tag: None)
Cloning into '/home/ziion/reports/tswap/5-t-swap-audit/lib/openzeppelin-contracts'...
fatal: unable to access 'https://github.com/openzeppelin/openzeppelin-contracts/': Could not resolve host: github.com
fatal: clone of 'https://github.com/openzeppelin/openzeppelin-contracts' into submodule path '/home/ziion/reports/tswap/5-t-swap-audit/lib/openzeppelin-contracts' failed
Error: 
git submodule exited with code 128
make: *** [Makefile:15: install] Error 1

What am I doing wrong?

**Comment by cromewar** [comment]
Seems the foundry had trouble to download from the Openzepellin repo, were you able to solve this @Himchenhim ?

**Reply by GenevaRoc** [reply]
how were you able to resolve the issue? coz I'm encountering this same error

**Comment by xaiksan1** [comment]
error code 128. 3 possible reason. 
1st: your SSH key is invalid or has been revoked. 
2nd: Incorrect Git configuration settings. Ensure that your Git username and email are properly configured:
 git config --global user.email "you@example.com"
3rd: and when i look at your path is probably this reason; Sometimes, the error occurs due to a long destination path where the project is being cloned. 
If it's none of this look your at git settings.

**Comment by akhilmanga** [comment]
Hi, @Himchenhim , Sometimes it happens due to a network issue. It happened to me while I'm cloning the `boss-bridge` audit.

Delete everything and start from scratch.

1. Clone the "5-t-swap-audit" repository.
2. Try to run the `make` command.

**Comment by EngrPips** [comment]
Primarily, This is caused by an unstable network.
